pkg/reconciler/pipelinerun/resources/pipelineref_test.go

```diff
@@ -529,71 +536,73 @@ func TestGetVerifiedPipelineFunc_Success(t *testing.T) {
 	}
 
 	testcases := []struct {
-		name                     string
-		requester                *test.Requester
-		resourceVerificationMode string
-		pipelinerun              v1beta1.PipelineRun
-		expected                 runtime.Object
+		name                      string
+		requester                 *test.Requester
+		verificationNoMatchPolicy string
+		pipelinerun               v1beta1.PipelineRun
+		policies                  []*v1alpha1.VerificationPolicy
+		expected                  runtime.Object
+		expectedSource            *v1beta1.ConfigSource
 	}{{
-		name:                     "signed pipeline with enforce policy",
-		requester:                requesterSigned,
-		resourceVerificationMode: config.EnforceResourceVerificationMode,
-		pipelinerun:              pr,
-		expected:                 signedPipeline,
-	}, {
-		name:                     "unsigned pipeline with warn policy",
-		requester:                requesterUnsigned,
-		resourceVerificationMode: config.WarnResourceVerificationMode,
-		pipelinerun:              pr,
-		expected:                 unsignedPipeline,
+		name:                      "enforce signed pipeline with matching policy pass verification",
+		requester:                 requesterMatched,
+		verificationNoMatchPolicy: config.FailNoMatchPolicy,
+		pipelinerun:               pr,
+		policies:                  vps,
+		expected:                  signedPipeline,
+		expectedSource:            matchPolicySource,
 	}, {
-		name:                     "signed pipeline with warn policy",
-		requester:                requesterSigned,
-		resourceVerificationMode: config.WarnResourceVerificationMode,
-		pipelinerun:              pr,
-		expected:                 signedPipeline,
+		name:                      "warn unsigned pipeline without matching policies",
```

Comment:
do we also need test cases for warn + ignore with matching policies and passed verification?

Reply:
ah yes, thanks! We do need, added